services/core/methods/api_v2/controllers/blocks.js

```diff
@@ -49,18 +45,6 @@ const getBlocks = async params => {
 
 		const response = await Core.getBlocks(params);
 
-		if (typeof params.blockId === 'string') {
-			if (isEmptyArray(response.data)) {
-				throw new NotFoundException(`Block ID ${params.blockId} not found.`);
```

Comment:
Consider throwing `NotFoundException` at the appropriate place within `services/core/shared/core/compat/sdk_v5/blocks.js`.